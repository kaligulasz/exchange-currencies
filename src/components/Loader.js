import React from 'react';
import styled from 'styled-components';
import { Spinner5 } from 'styled-icons/icomoon/Spinner5';

const Background = styled.div`
  background: ${props => props.theme.color.primaryGradient};
  position: absolute;
  height: 100%;
  width: 100%;
  display: flex;
  justify-content: center;
  z-index: 1;
`;

const StyledSpinner = styled(Spinner5)`
  width: 3.5rem;
  animation: spin 1.2s linear infinite;
  
  @keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
`;

const Loader = () => (
  <Background>
    <StyledSpinner />
  </Background>
);

export default Loader;
