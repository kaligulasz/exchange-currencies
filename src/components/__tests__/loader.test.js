import Loader from '../Loader';

describe('<Loader />', () => {
  it('renders loader', () => {
    const component = shallow(<Loader />);

    expect(component).toMatchSnapshot();
  });
});
