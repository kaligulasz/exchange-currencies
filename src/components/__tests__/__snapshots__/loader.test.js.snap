// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> renders loader 1`] = `
<Loader__Background>
  <Loader__StyledSpinner />
</Loader__Background>
`;
